server/tests/services/pageLog.test.js

```diff
@@ -0,0 +1,83 @@
+import { actionTypes } from '../../models/ticket';
+import { fixtures, build } from '../../utils/factories';
+import pageLogService from '../../services/pageLog';
+
+describe('## PageLogService', () => {
+  const userId = '123456789098765432123456';
+  let ticketId;
+
+  beforeEach((done) => {
+    build('ticket', fixtures.ticket())
+      .then((ticket) => {
+        ticketId = ticket.id;
+        done();
+      });
+  });
+
+  describe('# addAction()', () => {
+    context('with valid input parameters', () => {
+      it('adds an action successfully', (done) => {
+        pageLogService.addAction(ticketId, actionTypes.CREATED, userId)
+          .then((ticket) => {
+            expect(ticket.actions[0].actionTaken).to.equal(actionTypes.CREATED);
+            expect(ticket.actions[0].userId.toString()).to.equal(userId);
+            done();
+          });
+      });
+    });
+
+    context('with invalid input parameters', () => {
+      it('fails to add an action', (done) => {
+        pageLogService.addAction(ticketId, 'testing', userId)
+          .catch((err) => {
+            expect(err.details[0].message)
+              .to.equal('"actionType" must be one of [CREATED, PAGE_SENT, ACKNOWLEDGED, REJECTED, CLOSED]');
+            done();
+          });
+      });
+    });
+  });
+
+  describe('# removeAction()', () => {
+    let pageLogTimestamp;
+
+    beforeEach((done) => {
+      pageLogService.addAction(ticketId, actionTypes.CREATED, userId)
```

Comment:
I'm not a huge fan of using the service we're testing in the actual test itself. Can you instead use the fixtures helper to create a ticket that has the expected action on it?